### PR TITLE
vim rebuild sans nls

### LINKED ISSUE #5342: [vim_runtime] Cannot convert from "ISO-8859-1" to "UTF-8"
```
echo af ca cs cs.cp1250 da de en_GB eo es fi fr ga it ja ja.euc-jp ja.sjis ko ko.UTF-8 lv nb nl no pl pl.UTF-8 pl.cp1250 pt_BR ru ru.cp1251 sk sk.cp1250 sr sv tr uk uk.cp1251 vi zh_CN zh_CN.UTF-8 zh_CN.cp936 zh_TW zh_TW.UTF-8  | tr " " "\n" |sed -e '/\./d' | sort > LINGUAS
msgfmt --desktop -d . --template gvim.desktop.in -o tmp_gvim.desktop
msgfmt: Cannot convert from "ISO-8859-1" to "UTF-8". msgfmt relies on iconv(). This version was built without iconv().
make[2]: *** [Makefile:219: gvim.desktop] Error 1
make[2]: Leaving directory '/usr/local/tmp/crew/v8.2.2576.tar.gz.dir/vim-8.2.2576/src/po'
make[1]: *** [Makefile:2146: languages] Error 2
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/usr/local/tmp/crew/v8.2.2576.tar.gz.dir/vim-8.2.2576/src'
make: *** [Makefile:29: first] Error 2
vim_runtime failed to build: `make V=0 -j4` exited with 2
```

### PR DESCRIPTION
**Pull request by satmandu**
Fixes #5342

- issue was due to nls not working due to import from gettext which was no longer using libiconv.
- Maybe this is a problem with the older glibc not providing a good enough iconv? (or maybe vim just wants more for nls support?)
- solution was to just disable nls in the vim build.
- Also added a symlink for vi -> vim
- Finally, updated libcroco, which is a dep of gettext as per arch.
- using shiny new preflight

Works properly:
- [x] x86_64
- [x] armv7l
- [x] i686

### CODE REVIEW

File: packages/vim_runtime.rb [hunk 1]
```diff
@@ -3,49 +3,66 @@
...
-     x86_64: '588ce10bf392d363ce9de709f763ab79ef35a5e8e5eb67e278eec894b6fa04ed',
+  binary_sha256({
+    aarch64: '89f2375c66810417550d4ab2d2fbf90a6953e3c585b8e0baaa1240b7e0187c99',
+     armv7l: '89f2375c66810417550d4ab2d2fbf90a6953e3c585b8e0baaa1240b7e0187c99',
+       i686: '0b7505d98e974e45d621f33fcaa19f14d907422d3b04f41bfdb0d9143776d61d',
+     x86_64: 'a2ee9fcb92bc566200d21a176ea7bfec6ef7d5b3fbe3fc3c0244ed31e6168d14'
   })
 
+  def self.preflight
+    raise StandardError, 'Please remove libiconv before building.' if File.exist?("#{CREW_LIB_PREFIX}/libcharset.so")
```
Comment (supechicken) [inline packages/vim_runtime.rb:26]: Better to use `abort`? (Like a warning, not an runtime error)

Comment (satmandu) [reply packages/vim_runtime.rb]: Does this look better?
`abort('Please remove libiconv before building.') if File.exist?("#{CREW_LIB_PREFIX}/libcharset.so")`

File: packages/vim.rb [hunk 2]
```diff
@@ -3,61 +3,79 @@
...
   depends_on 'vim_runtime'
 
+  def self.preflight
+    abort('Please remove libiconv before building.') if File.exist?("#{CREW_LIB_PREFIX}/libcharset.so")
+  end
+
   def self.preinstall
     gvim = `which gvim 2> /dev/null`.chomp
-    abort "gvim version #{version} already installed.".lightgreen unless "#{gvim}" == ""
+    abort "gvim version #{version} already installed.".lightgreen unless gvim.to_s == ''
```
Comment (uberhacker) [inline packages/vim.rb:33]: Since this is only checking for the version, you can move this to `def self.preflight`.

**Comment by uberhacker** [comment]
Should we include `gvim` also?

**Comment by satmandu** [comment]
gvim would also be a good idea.

**Comment by satmandu** [comment]
building gvim for this PR...

**Comment by satmandu** [comment]
Done.

**Pull request merged**